### ISSUE TITLE
Pip install leads to cjoin errors on import

### ISSUE THREAD

**Issue by bmschmidt** [TomScheffers/pyarrow_ops#1]
Interesting project--found it from the datafusion issues page, and looks useful. Thanks for sharing!

As an FYI: `pip install` works on my computer (Mac OS X, python 3.8.5), but then import of 'join' from this module fails with an error message about "expected 80 bytes" or something in cjoin.c. (Apologies, I've lost the bad message; but the points was the Cython generated cjoin.c file didn't run on my system.)

I was able to successfully install the package by cloning and running `python setup.py build; python setup.py install`.

**Comment by TomScheffers** [comment]
@bmschmidt Good to hear you are interested in the project. I am currently working on datafusion like functionality in python, [which you can find here](https://github.com/TomScheffers/wombat) (the pyarrow_ops functionality is included). 

Regarding the issue in the pip installation, I am unable to replicate the issue (I use Mac OS X and python 3.8.8 myself). Unfortunately I am not a pypi wizard: I was unable to include the cython packages directly and make `pip install` work on my desktop. Therefore I added the cjoin.c files directly.

If anyone has knowledge about how to improve the packaging, please let me know!

**Comment by bmschmidt** [comment]
Super, will check out the wombat project. I have very little understanding of either pip or cython, but fiddling for a minute on my laptop (OS X 10.14, 3.7.7) I get the same error, so at least I can paste it below! It applies only to the 'join' import.

```

% echo "from pyarrow_ops import join" | python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/bschmidt/opt/miniconda3/lib/python3.7/site-packages/pyarrow_ops/__init__.py", line 4, in <module>
    from pyarrow_ops.join import join
  File "/Users/bschmidt/opt/miniconda3/lib/python3.7/site-packages/pyarrow_ops/join.py", line 5, in <module>
    from cjoin import inner_join
  File "pyarrow_ops/cjoin.pyx", line 1, in init pyarrow_ops.cjoin
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
```

**Comment by TomScheffers** [comment]
@bmschmidt the issue was that I included the cjoin.c file directly, which only works if you have exactly the same version of GCC. Therefore, I had to cythonize the extension in the setup.py instead. I fixed this in version 0.0.8, let me know if it works for you as well.

Thanks for your help!
